packages/sakuli-legacy/src/context/common/test-case/test-case.class.ts

```diff
@@ -28,6 +34,29 @@ export function createTestCaseClass(
       readonly criticalTime: number = 0,
       public _imagePaths: string[] = []
     ) {
+      throwOnRuntimeTypeMissmatch(
+        caseId,
+        "String",
+        `Parameter caseId is invalid, string expected. Value: ${caseId}`
+      );
+      throwOnRuntimeTypeMissmatch(
+        warningTime,
+        "Number",
+        `Parameter warningTime is invalid, number expected. Value: ${warningTime}`
+      );
+      throwOnRuntimeTypeMissmatch(
+        criticalTime,
+        "Number",
+        `Parameter criticalTime is invalid, number expected. Value: ${criticalTime}`
+      );
+      for (const imagePath of _imagePaths) {
+        throwOnRuntimeTypeMissmatch(
+          imagePath,
+          "String",
+          `Parameter _imagePaths is invalid, string expected. Value: ${imagePath}`
+        );
+      }
+
```

Comment:
Even if this is nothing to be reused, in my opinion it would make sense to extract this into a function to encapsulate the parameter validation and limit the visible details of the scope of the constructor.